### DISCUSSION TITLE
Use KDE fork for Qt5?

### DISCUSSION THREAD

**Discussion by Be-ing** [microsoft/vcpkg#20115] [Ideas]
Since the Qt Company has refused to release maintenance updates to the last Qt5 release except to customers paying for proprietary licenses, KDE is [maintaining a fork of Qt 5.15.2](https://dot.kde.org/2021/04/06/announcing-kdes-qt-5-patch-collection) with patches that have been approved upstream for Qt6 backported to Qt 5.15.2. I think it may be a good idea for vcpkg to package KDE's fork in the Qt 5.15.2 ports. Qt6 is still quite new and for many large projects it will take a long time to migrate to Qt6. In the meantime, it would be nice to use the bug and security fixes from the KDE fork. Would that be welcome in vcpkg?

**Comment by Neumann-A** [comment]
Feel free to do that. But it is manual updating over all ports since the KDE fork does not tag versions which is really annoying. 
The `latest` feature in `qt5-base` should be used to implement the KDE version but it requires a lot of work and changes to make it work for both. I have some preliminary work in https://github.com/Neumann-A/vcpkg/tree/qt_kde but don't have the time or the motivation to finish that. I just personally use Qt6 and stuff like VTK can be switched to it already.

**Reply by Be-ing** [reply]
I probably won't do the work unless Mixxx happens to be affected by a bug which got fixed in the KDE fork on Windows (we cannot use Qt 5.15 on macOS due to severe performance regressions which are unrelated to vcpkg). But I thought it would be a good idea to open the discussion in case someone else wants to pick up this work.

**Reply by wrobelda** [reply]
> Feel free to do that. But it is manual updating over all ports since the KDE fork does not tag versions which is really annoying.

IIRC, this may have been part of the agreement between Trolltech and KDE. However, instead of actually switching to KDE fork, we could pull in all the patches instead. They are, however, 7MB in size combined...

**Comment by Be-ing** [comment]
Ping @wrobelda @nicolasfella
